### PR TITLE
convert prefetched values generator to reusable iterator

### PR DESCRIPTION
**Pull request by TimWhalen**
One way to address the issue I opened last week: https://github.com/graphite-project/graphite-web/issues/2318

### CODE REVIEW

**Comment by DanCech** [comment]
Need to dig into this a bit and make sure there's no performance regression

**Comment by iksaif** [comment]
I'm pretty sure this *will* have performance regressions. Doing it in two loops (first getting all the generators, then expanding them) would maybe work.

**Comment by TimWhalen** [comment]
Thanks, I'll move the generator expansion to render.evaluate.evaluateTargets and
I'll try to add a meaningful test too.  If only the rrdtool backend is returning values in
a generator maybe it'd be better to address this there?

**Comment by iksaif** [comment]
FYI: It's not only RRD, it's also a bunch for third-party readers (mostly biggraphite). I won't have a stable  internet connection in the next three weeks so I can't validate your patch, but it looks like it would work (maybe add some code comments to explain why it is done this way).

Thanks

**Comment by deniszh** [comment]
Is it OK to merge that? @DanCech @iksaif ?

**Comment by iksaif** [comment]
I think it's still missing comments to explain why it is done this way. Else it's likely that the code will be changed again at some point.

**Comment by TimWhalen** [comment]
I've added a brief comment explaining the change.

File: webapp/graphite/render/datalib.py [hunk 1]
```diff
@@ -271,6 +272,13 @@ def prefetchData(requestContext, pathExpressions):
       ),
     ))
 
+  # Several third-party readers including rrdtool and biggraphite return values in a
+  # generator which can only be iterated on once. These must be converted to a list.
+  for pathExpression, items in prefetched.items():
+    for i, (name, (time_info, values)) in enumerate(items):
+      if isinstance(result['values'], types.GeneratorType):
```
Comment (DanCech) [inline webapp/graphite/render/datalib.py]: where is `result['values']` coming from? assuming this should be `values`

Comment (TimWhalen) [reply webapp/graphite/render/datalib.py]: You are correct, that was a mistake introduced when I moved the check into its own loop to address an earlier change request.  Sorry about that.  I've updated the pr again.

**Pull request merged**